client/src/app/site/polls/services/poll.service.ts

```diff
@@ -267,11 +267,14 @@ export class PollService {
     }
 
     private sumOptionsYN(option: OptionData): number {
-        return (option?.yes ?? 0) + (option?.no ?? 0);
+        const yes = option?.yes ?? 0;
+        const no = option?.no ?? 0;
+        return (yes >= 0 ? yes : 0) + (no >= 0 ? no : 0);
     }
 
     private sumOptionsYNA(option: OptionData): number {
-        return this.sumOptionsYN(option) + (option?.abstain ?? 0);
+        const abstain = option?.abstain ?? 0;
+        return this.sumOptionsYN(option) + (abstain >= 0 ? abstain : 0);
```

Comment:
Why did you change it?

Reply:
Because I found out that if values were set to VOTE_UNDOCUMENTED (i.e. set to -2), they would be added onto the sum, even if they really shouldn't.
The result was that the sum would be (sum of documented votes - 2*number of undocumented options).
Since percentage values are calculated on these functions this would warp them (sometimes to the point where certain options would have values like 200%).
To counteract this I changed the function to not add on values that are smaller than 0